src/gentropy/locus_breaker_clumping.py

```diff
@@ -0,0 +1,72 @@
+
+from __future__ import annotations
+
+from gentropy.common.session import Session
+from gentropy.dataset.summary_statistics import SummaryStatistics
+from gentropy.method.locus_breaker_clumping import LocusBreakerClumping
+
+
+class LocusBreakerClumpingStep:
+    """Step to perform locus-breaker clumping on a study."""
+
+    def __init__(
+        self,
+        session: Session,
+        summary_statistics_input_path: str,
+        clumped_study_locus_output_path: str,
+        lbc_baseline_pvalue: float,
+        lbc_distance_cutoff: int,
+        lbc_pvalue_threshold: float,
+        lbc_flanking_distance: int,
+        large_loci_size: int,
+        wbc_pvalue_threshold: float,
+        collect_locus_distance: int,
+        collect_locus: bool = False,
+        remove_mhc: bool = True,
+    ) -> None:
+        """Run locus-breaker clumping step.
+
+        This step will perform locus-breaker clumping on the full set of summary statistics.
+        StudyLocus larger than the large_loci_size, by distance, will be further clumped with window-based
+        clumping.
+
+        Args:
+            session (Session): Session object.
+            summary_statistics_input_path (str): Path to the input study locus.
+            clumped_study_locus_output_path (str): path of the resulting, clumped study-locus dataset.
+            lbc_baseline_pvalue (float): Baseline p-value for locus breaker clumping.
+            lbc_distance_cutoff (int): Distance cutoff for locus breaker clumping.
+            lbc_pvalue_threshold (float): P-value threshold for locus breaker clumping.
+            lbc_flanking_distance (int): Flanking distance for locus breaker clumping.
+            large_loci_size (int): Threshold distance to define large loci for window-based clumping.
+            wbc_pvalue_threshold (float): P-value threshold for window breaker clumping.
+            collect_locus_distance (int): Distance to collect locus.
+            collect_locus (bool, optional): Whether to collect locus. Defaults to False.
+            remove_mhc (bool, optional): If true will use exclude_region() to remove the MHC region.
+        """
+        sum_stats = SummaryStatistics.from_parquet(
+            session, summary_statistics_input_path
+        )
+        lbc = sum_stats.locus_breaker_clumping(
+            lbc_baseline_pvalue,
+            lbc_distance_cutoff,
+            lbc_pvalue_threshold,
+            lbc_flanking_distance,
+        )
+        clumped_result = LocusBreakerClumping._process_locus_breaker(
+            lbc,
+            sum_stats,
+            large_loci_size,
+            wbc_pvalue_threshold,
+        )
+        if collect_locus:
+            clumped_result = clumped_result.annotate_locus_statistics(
+                sum_stats, collect_locus_distance
+            )
+        if remove_mhc:
+            clumped_result = clumped_result.exclude_region("chr6:25726063-33400556")
```

Comment:
I would move this bit of logic earlier, so if HMC is not desired in the output, we won't actually do the clumping on on this region. But worth checking if the filtering is faster or slower than the actual clumping.

Reply:
Removing MHC region from the sum stats before clumping can lead to some problems. If there is a very strong signal within the region, variants flanking the region will create potentially false peaks/loci.
![image](https://github.com/opentargets/gentropy/assets/113430683/c701b4bf-d042-44c8-b250-0ba03d88de0c)
![image](https://github.com/opentargets/gentropy/assets/113430683/110cf886-c828-48de-9108-016a3ba15d5b)
Removing loci after clumping which overlap the xMHC region by locusStart/locusEnd deals with these scenarios pretty well.

Reply:
Ah, OK so this way you sort out signal "overflow". Makes sense.